### PR TITLE
Added to syslog_output functionality to resolve hostnames

### PR DESCRIPTION
**Pull request by jotacarma90**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/4227|

## Description
This PR adds the option to configure syslog_output with a hostname instead of the IP.
I removed the chroot from the **csyslogd** daemon, used the `resolveHostname` function to get the IP, and created a function, `get_ip_from_resolvedHostname`, that extracts the IP from the format returned by `resolveHostname`.
I also added extra functionality in `OS_Alert_SendSyslog` and `OS_Alert_SendSyslog_JSON` to try to reconnect to the socket at least one more time if it shows as disconnected.

## Configuration options
Now, its able to configure a hostname, for example:
```
<syslog_output>
  <server>localhost</server>
  <level>7</level>
  <format>json</format>
</syslog_output>

```

## Logs/Alerts example
`2021/03/17 08:54:44 wazuh-csyslogd[19382] csyslogd.c:85 at OS_CSyslogD(): DEBUG: Resolving server hostname: localhost
`
## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
- [x] Source installation
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)

### CODE REVIEW

File: src/shared/os_utils.c [hunk 1]
```diff
@@ -177,6 +179,44 @@ int w_del_plist(OSList *p_list)
...
+        free(f_ip);
+        free(*hostname);
+        os_strdup(ip_str, *hostname);
+    } else {
+        char ip_str[128] = {0};
+        snprintf(ip_str, 127, "%s/", *hostname);
+        free(*hostname);
+        os_strdup(ip_str, *hostname);
+    }
+}
```
Comment (antoniomanuelfr) [inline src/shared/os_utils.c:209]:
```suggestion
    char ip_str[128] = {0};
    if (f_ip) {
        snprintf(ip_str, 127, "%s/%s", *hostname, f_ip);
        free(f_ip);
    } else {
        snprintf(ip_str, 127, "%s/", *hostname);
    }
    free(*hostname);
    os_strdup(ip_str, *hostname);
}

```

**Review by antoniomanuelfr — APPROVED** [review]
LGTM

**Review by Molter73 — CHANGES_REQUESTED** [review]
`src/wazuh-logtest-legacy` should not be part of this PR, please remove it.

We should also add unit tests to `resolveHosname` and `get_ip_from_resolved_hostname`.

**Review by Molter73 — APPROVED** [review]
LGTM!

**Review by antoniomanuelfr — APPROVED** [review]

**Pull request merged**